go-controller/pkg/ovn/service.go

```diff
@@ -169,6 +169,14 @@ func (ovn *Controller) createService(service *kapi.Service) error {
 		return nil
 	}
 
+	// We can end up in a situation where the endpoint creation is triggered before the service creation,
+	// we should not be creating the reject ACLs if this endpoint exists, because that would result in an unreachable service
+	// eventough the endpoint exists.
+	ep, err := ovn.watchFactory.GetEndpoint(service.Namespace, service.Name)
```

Comment:
if someone creates a service with a pod selector that has no matches, a corresponding EP will still be created that contains no addresses. With this code we wont setup the rejectACL appropriately I think. We need to also check if ep.Subsets length is greater than 0. If we want to be super careful, we could also check that a subset.Addresses >0 as well.

Reply:
FYI: I didn't add a check for `subset.Addresses > 0` because I tested it and saw that services has a sort of validating admission webhook, that states:

```
subsets[0]: Required value: must specify `addresses` or `notReadyAddresses`
```

when you remove all addresses from a subset. So subsets always have addresses.